### PR TITLE
Add more details to the Job error message

### PR DESCRIPTION
**Pull request by fao89**

### CODE REVIEW

File: modules/common/condition/conditions.go [hunk 1]
```diff
@@ -103,6 +103,10 @@ const (
 	// This is a warning because the reconciler will retry deletion.
 	ErrorReason = "Error"
 
+	// ErrorReason (Severity=Error) documents a condition not in Status=True because the underlying object reached the backoff limit.
```
Comment (stuggi) [inline modules/common/condition/conditions.go:106]: `ErrorReason` -> `JobReasonBackoffLimitExceeded`

File: modules/common/job/job.go [hunk 2]
```diff
@@ -207,6 +208,22 @@ func (j *Job) GetHash() string {
 	return j.hash
 }
 
+// GetTotalAttempts func
+func (j *Job) GetTotalAttempts() int64 {
+	if j.actualJob == nil {
+		return 0
+	}
+	return int64(j.actualJob.Status.Failed)
```
Comment (stuggi) [inline modules/common/job/job.go:216]: should this be `Status.Failed` + `Status.Succeeded` ? like the total attempts of a job would be e.g.  if you have 2 failed and last succeeded so in total you'd have 2 + 1.  Or should rename to `GetTotalFailedAttempts` ?

Comment (fao89) [reply modules/common/job/job.go:216]: I wanted to register the retries, so renamed to `GetTotalFailedAttempts`

**Review by jpodivin — CHANGES_REQUESTED** [review]

File: modules/common/job/job.go [hunk 3]
```diff
@@ -207,6 +208,22 @@ func (j *Job) GetHash() string {
 	return j.hash
 }
 
+// GetTotalFailedAttempts func
+func (j *Job) GetTotalFailedAttempts() int64 {
+	if j.actualJob == nil {
+		return 0
+	}
+	return int64(j.actualJob.Status.Failed)
```
[the remark this answers is not in the document]
Comment (fao89) [reply modules/common/job/job.go]: for passing it to `strconv.FormatInt`, but you are right, changing the `fmt` formatting is better

**Review by jpodivin — COMMENTED** [review]
+2 From me. If nobody else has any suggestions we can merge this.

**Comment by stuggi** [comment]
we are in merge freeze. lets not merge things until then

**Comment by jpodivin** [comment]
> can we unhold and merge now?

I think we can.

**Review by jpodivin — APPROVED** [review]

**Pull request merged**